src/cpp/TextNote.cpp

```diff
@@ -56,4 +56,29 @@ TEST_CASE("creating object with fixed creation time") {
 
 }
 
+TEST_CASE("Match with empty criteria") {
+    TextNote item("a");
+    SearchCriteria<std::string, std::time_t> criteria;
+    CHECK(item.match(&criteria) == 1);
+}
+
+TEST_CASE("Match with criteria with non-matching conditions") {
+    TextNote item("a");
+    SearchCriteria<std::string, std::time_t> criteria;
+    criteria.addEqualToCheck<std::string>("title", "nine");
+    CHECK(item.match(&criteria) == 0);
+}
 
+TEST_CASE("Match with criteria with matching condition") {
+    TextNote item("a");
+    SearchCriteria<std::string, std::time_t> criteria;
+    criteria.addEqualToCheck<std::string>("title", "a");
+    CHECK(item.match(&criteria) == 1);
+}
+TEST_CASE("Match with criteria with matching and non-matching conditions") {

```

Comment:
TextNote has no implementation of match that takes into account its text property?
And no unit tests for this?